src/main/java/pl/ug/virtualofficebackend/domain/user/boundary/UserService.java

```diff
@@ -0,0 +1,4 @@
+package pl.ug.virtualofficebackend.domain.user.boundary;
+
+public interface UserService {
```

Comment:
Czy boundary ma byc folderem w ktorym bedziemy trzymac tylko interfejs?

Reply:
Jeśli zrobimy oddzielne serwisy(które będzą interfejsami) dla logowania i rejestracji, to one również miałyby być w boundary.
Czyli wszystkie interfejsy, do których chcemy dawać dostęp innym osobom.

Reply:
Chcemy logowanie i rejestracje trzymac jako oddzielne serwisy? Imo troche overkill i mozna to spokojnie trzymac w jednym

Reply:
W teorii powinniśmy w dwóch oddzielnych, ale możemy też trzymać w jednym. Co w sumie raczej nie zmienia faktu, że logowanie i rejestracja nie powinna być w tym samym co User.
Ogólnie ten commit jest do pokazania tej architektury, więc raczej trzeba będzie zmieniać dodać nowe klasy do logowania itd

Reply:
Plus czy  w takim wypadku beda tutaj tylko interfejsy serwisow? Czy w przypadku walidatora np. bedziemy to trzymac w 2 odzielnych pakietach?

Reply:
Tak, tylko interfejsy serwisów.
No jeżeli walidatory byłyby tylko użyte w tym pakiecie, to wrzucamy normalnie do internal.
Jeśli w różnych pakietach, to wydzieliłbym oddzielny pakiet do samych walidatorów, jak np.
domain/validator

Reply:
W takim razie jak chcemy robic rejestracje? Imo to raczej powinno byc w pakiecie User, bo tutaj bedziemy dodawac/usuwac nowych userow i ich walidowac

Reply:
domain/boundary - LoggingService
domain/boundary - UserService
domain/internal - LoggingServiceImpl
domain/internal - UserServiceImpl
domain/internal - UserValidator

Reply:
Czym różni sie LogginService od UserService? Samo Logging service bedzie tylko hashowaniem i porownywaniem passow do tych w bazie wiec imo to mozna spokojnie dodac do UserService

Reply:
Myślałem, że jeśli bedziemy w przyszłości stworzyć tokeny JWT i na ich podstawie robić autoryzację, to będzie potrzebny oddzielny serwis do tego.
Jeśli chcemy to zrobić później, to możemy to na razie olać i dodac do UserService.
Pisząc to myślałem już o reście, bo w nim już powinny być dwa oddzielne kontrolery dla usera i dla logowania się.

Reply:
Hmm... no spoczko. Ja bardziej podchodze do tego w taki sposob zeby sie wyrobic ze wszystkim w ten miesiac :D